client-js/app/scripts/obligation.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Obligation is the input side complement of Promise; if one party is promised
+ * a value, then of necessity another party must be obliged to provide the
```

Comment:
"of necessity" = 26.5 M google results; "by necessity" = 18.8 M google results. Therefore "of necessity" wins :)

But yes, I like your wording better anyway.